dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -153,6 +162,23 @@ export function AppRepoForm(props: IAppRepoFormProps) {
       );
       setValidated(currentlyValidated);
     }
+    let filter: IAppRepositoryFilter | undefined;
+    if (type === TYPE_HELM && filterNames !== "") {
+      if (filterRegex) {
+        filter = { jq: ".name | test($var)", variables: { $var: filterNames } };
+      } else {
+        const names = filterNames.split(",").map(n => n.trim());
+        const variables = names.reduce((acc, n, i) => {
+          acc[`$var${i}`] = n;
+          return acc;
+        }, {});
+        const jq = names.map((v, i) => `.name == $var${i}`).join(" or ");
```

Comment:
Not sure, but perhaps does it worth extracting this logic to a function in charge of generating the `jq` ?